lib/record_store/zone.rb

```diff
@@ -32,10 +32,10 @@ def download(name, provider_name, **write_options)
         zone.write(**write_options)
       end
 
-      def filter_records(current_records, ignore_patterns)
-        ignore_patterns.inject(current_records) do |remaining_records, pattern|
+      def filter_records(current_records, ignore_objects)
+        ignore_objects.inject(current_records) do |remaining_records, pattern|
           remaining_records.reject do |record|
-            pattern.all? { |(key, value)| record.respond_to?(key) && value === record.send(key) }
+            pattern.should_ignore?(record)
```

Comment:
Nitpick... since the calls are pretty short, we could just make this a one-liner like so:

```ruby
remaining_records.reject { |record| pattern.should_ignore?(record) }
```